src/lib/viewers/media/DashViewer.js

```diff
@@ -387,6 +406,25 @@ class DashViewer extends VideoBaseViewer {
         }
     }
 
+    /**
+     * Handles streaming event which is the first time the manifest is available. See https://shaka-player-demo.appspot.com/docs/api/shaka.util.Error.html
+     *
+     * @private
+     * @param {Object} shakaError - Error to handle
+     * @return {void}
+     */
+    shakaManifestHandler() {
```

Comment:
I only moved things that are dependent on the manifest here. This should give a slight performance gain as we will zero in on our target quality faster than if we waited for `loadeddata`.

Reply:
💯 I should look into adding startAt here as well. I think there was a bug which I had to work around before but pretty sure they fixed it